### PR TITLE
Function search

### PR DESCRIPTION
**Pull request by waynodiablo**
Added feature allowing user to display data by filtering by Function using a dropdown menu.

### CODE REVIEW

**Review by SBriscoe2 — APPROVED** [review]
All amendments working as expected.
Changes approved.

**You may wish to alter the html text re: Pick a function to something more user friendly**

File: workforceDashboard/src/main/java/gov/homeoffice/workforceDashboard/repository/EmployeeRepository.java [hunk 1]
```diff
@@ -8,8 +8,19 @@
...
 
 public interface EmployeeRepository extends JpaRepository<Employee, String> {
-    //  WS - The below is the code to return a sorted list by Function
-    @Query("SELECT a FROM Employee a ORDER BY a.function ASC, a.team ASC, a.employeeFullName ASC")
+
+    @Query("SELECT a FROM Employee a ORDER BY a.function ASC, a.team ASC, a.employeeSurname ASC")
     List<Employee> findByFunction();
+
+    @Query("SELECT DISTINCT f.function FROM Employee f ORDER BY f.function ASC")
+    List<String> fieldSelect();
```
Comment (catherine-nakka) [inline workforceDashboard/src/main/java/gov/homeoffice/workforceDashboard/repository/EmployeeRepository.java]: The function name has to reflect what its exactly trying to do. So far unless I see the query  in the repository its unclear that the method is getting a list of unique functions.

File: workforceDashboard/src/main/java/gov/homeoffice/workforceDashboard/repository/EmployeeRepository.java [hunk 2]
```diff
@@ -8,8 +8,19 @@
...
+
+    @Query("SELECT s FROM Employee s WHERE s.function= ?1 ORDER BY s.employeeSurname ASC")
+    List<Employee> findBySelection(String selection);
```
Comment (catherine-nakka) [inline workforceDashboard/src/main/java/gov/homeoffice/workforceDashboard/repository/EmployeeRepository.java]: Method name needs changing to exactly reflect its functionality

File: workforceDashboard/src/main/java/gov/homeoffice/workforceDashboard/repository/EmployeeRepository.java (hunk 1 above)
Comment (waynodiablo) [reply workforceDashboard/src/main/java/gov/homeoffice/workforceDashboard/repository/EmployeeRepository.java]: Changed to getUniqueFunctions()

Comment (waynodiablo) [reply workforceDashboard/src/main/java/gov/homeoffice/workforceDashboard/repository/EmployeeRepository.java]: Chnaged again, this time to getListOfUniqueFunctions()

File: workforceDashboard/src/main/java/gov/homeoffice/workforceDashboard/repository/EmployeeRepository.java (hunk 2 above)
Comment (waynodiablo) [reply workforceDashboard/src/main/java/gov/homeoffice/workforceDashboard/repository/EmployeeRepository.java]: Chnaged to findByUniqueFunction()

**Comment by waynodiablo** [comment]
Comments resolved, also made change to HTML text.

**Comment by waynodiablo** [comment]
Removed a stray logger.

**Pull request merged**